lib/report_portal/cucumber/parallel_report.rb

```diff
@@ -35,65 +35,65 @@ def initialize(desired_time)
         ReportPortal.last_used_time = 0
         set_parallel_tests_vars
         if ParallelTests.first_process?
-          File.open(file_with_launch_id, 'w') do |f|
-            f.flock(File::LOCK_EX)
-            start_launch(desired_time, @cmd_args_of_parallel_tests)
-            f.write(ReportPortal.launch_id)
-            f.flush
-            f.flock(File::LOCK_UN)
-          end
-          @folder_creation_tracking_file = Pathname(Dir.tmpdir)  + "folder_creation_tracking_#{ReportPortal.launch_id}.lck"
-          File.open(@folder_creation_tracking_file, 'w+') do |f|
-            f.flock(File::LOCK_EX)
-            f.flush
-            f.flock(File::LOCK_UN)
-          end
+          start_launch(desired_time)
         else
           start_time = monotonic_time
           loop do
-            break if File.exist?(file_with_launch_id)
+            break if File.exist?(lock_file)
             if monotonic_time - start_time > wait_time_for_launch_start
               raise "File with launch ID wasn't created after waiting #{wait_time_for_launch_start} seconds"
             end
             sleep 0.5
           end
-          File.open(file_with_launch_id, 'r') do |f|
-            f.flock(File::LOCK_SH)
+          File.open(lock_file, 'r') do |f|
             ReportPortal.launch_id = f.read
-            f.flock(File::LOCK_UN)
           end
-          sleep_time = ENV['TEST_ENV_NUMBER'].to_i
+          add_process_description
+          sleep_time = 5
           sleep(sleep_time) # stagger start times for reporting to Report Portal to avoid collision
         end
       end
 
+      def add_process_description
+        description = ReportPortal.get_launch['description'].split(' ')
+        description.push(self.description().split(' ')).flatten!
+        ReportPortal.update_launch({description: description.uniq.join(' ')})
+      end
+
       def done(desired_time = ReportPortal.now)
         end_feature(desired_time) if @feature_node
 
         if ParallelTests.first_process?
           ParallelTests.wait_for_other_processes_to_finish
 
-          File.delete(file_with_launch_id)
+          File.delete(lock_file)
 
-          unless attach_to_launch?
-            $stdout.puts "Finishing launch #{ReportPortal.launch_id}"
+          # TODO: if parallel test does not need to finish launch, there should be env variable to support this.
```

Comment:
The idea of `attach_to_launch` was different.
The idea was to create a launch before starting execution of the tests.
And don't end it when tests finish execution.
Tests only report to that launch without starting or finishing it.

The previous behavior has a use case:
A launch is created. Tests are divided into portions. Each portion is executed on a separate computer. Results from each portion are reported to the same launch. Command to finish launch is sent only when all portions finish.

Reply:
This behavior will still work, I did not change the priority:
https://github.com/reportportal/agent-ruby/blob/e62541eb7242e3e454c029161ad7d28d2a89bebd/lib/report_portal/cucumber/report.rb#L57

But from a user point of view, you simply want to run tests as before without extra API client to manage report launches

Reply:
Yes, you didn't change the priority when starting the launch.
But previously if the launch id was passed via env variable, launch were not closed after tests finish. It seems that behavior was changed.

Reply:
this should fix it:
https://github.com/reportportal/agent-ruby/blob/01fbe3a8261194f55cc96029595266583875fc54/lib/report_portal/cucumber/parallel_report.rb#L74